consensus/src/sync/light/sync_requests.rs

```diff
@@ -0,0 +1,267 @@
+            } else {
+                // Check that the epoch_id sent by the peer at our current epoch number corresponds to
+                // our accepted state. If it doesn't, the peer is on a "permanent" fork, so we ban it.
+                let peers_epoch_id =
+                    &epoch_ids.ids[our_epoch_number - epoch_ids.first_epoch_number];
+                if our_epoch_id != *peers_epoch_id {
+                    // TODO Actually ban the peer.
+                    debug!(
+                        our_epoch_number,
+                        %our_epoch_id,
+                        %peers_epoch_id,
+                        peer = %epoch_ids.sender,
+                        "Peer is on a different chain"
+                    );
+                    return Some(epoch_ids.sender);
+                }
+
+                epoch_ids.ids = epoch_ids
+                    .ids
+                    .split_off(our_epoch_number - epoch_ids.first_epoch_number + 1);
+                epoch_ids.first_epoch_number = our_epoch_number + 1;
+            }
+        }
+
+        // Discard checkpoint block if it is old.
+        if let Some(checkpoint) = &epoch_ids.checkpoint {
+            if checkpoint.block_number <= our_block_number {
+                epoch_ids.checkpoint = None;
+            }
+        }
+
+        let mut peer_requests = PeerMacroRequests::new();
+
+        // Request the election blocks
+        for block_hash in epoch_ids.ids {
+            let network = Arc::clone(&self.network);
+            let peer_id = epoch_ids.sender;
+
+            peer_requests.push_request(block_hash.clone());
+
+            self.block_headers.push(
+                async move {
+                    (
+                        Self::request_block(network, peer_id, block_hash).await,
+                        peer_id,
+                    )
+                }
+                .boxed(),
+            );
+        }
+
+        // Request the checkpoint (if any)
+        if let Some(checkpoint) = &epoch_ids.checkpoint {
+            let block_hash = checkpoint.clone().hash;
+            let network = Arc::clone(&self.network);
+            let peer_id = epoch_ids.sender;
+            peer_requests.push_request(block_hash.clone());
+            self.block_headers.push(
+                async move {
+                    (
+                        Self::request_block(network, peer_id, block_hash).await,
+                        peer_id,
+                    )
+                }
+                .boxed(),
+            );
+        }
+
+        self.peer_requests.insert(epoch_ids.sender, peer_requests);
+
+        None
+    }
+
+    pub async fn request_block(
+        network: Arc<TNetwork>,
+        peer_id: TNetwork::PeerId,
+        hash: Blake2bHash,
+    ) -> Result<Option<Block>, RequestError> {
+        network
+            .request::<RequestBlock>(
+                RequestBlock {
+                    hash,
+                    include_body: true,
```

Comment:
I'd put a comment that since these are macro blocks, we do want to include the body